### PR TITLE
feat: add entries summary section with comprehensive tests

### PR DESCRIPTION
**Pull request by JuanVqz**
## Summary

This PR adds a new summary section to the EntriesView that displays time tracking statistics and includes comprehensive test coverage.

### ✨ New Features
- **Entries Summary Section**: Added a summary bar that displays:
  - Total time, billable time, and unbillable time
  - Entry count with proper pluralization
  - Billable percentage calculation
  - Visual indicators with colored icons (green for billable, red for unbillable)

### 🧪 Testing
- **Comprehensive Test Coverage**: Added 7 test cases for the `calculateEntrySummary` function:
  - Mixed billable and unbillable entries
  - All billable entries scenario
  - All unbillable entries scenario
  - Empty entries array handling
  - Single entry handling
  - Zero minutes entries
  - Large time values (24+ hours)

### 🔧 Technical Details
- New `calculateEntrySummary` utility function in `utils.ts`
- Proper TypeScript interfaces for `EntrySummary`
- Memoized calculations for performance
- Conditional rendering based on entry count
- Proper time formatting using existing `hoursFormat` function

## Testing
- ✅ All existing tests pass (62 total)
- ✅ New tests cover edge cases and business logic
- ✅ No linting errors
- ✅ Proper TypeScript types

## Screenshots
The summary section appears at the top of the entries list when there are entries to display, showing:
- Total: 03:00 | Billable: 02:00 | Unbillable: 01:00
- 3 entries • 67% billable
- Visual icons for billable (green coins) and unbillable (red minus) time

This enhancement improves the user experience by providing quick insights into time tracking patterns without needing to manually calculate totals.

### CODE REVIEW

File: src/views/EntriesView.tsx [hunk 1]
```diff
@@ -76,7 +84,46 @@ export const EntriesView = ({ onCancel }: EntriesViewProps) => {
...
       ) : (
-        entryItems
+        <>
+          {entrySummary && entrySummary.entryCount > 0 && (
+            <List.Section title="Summary">
+              <List.Item
+                title={`Total: ${entrySummary.totalFormatted} | Billable: ${entrySummary.billableFormatted} | Unbillable: ${entrySummary.unbillableFormatted}`}
+                subtitle={`${entrySummary.entryCount} ${entrySummary.entryCount === 1 ? 'entry' : 'entries'} • ${entrySummary.totalMinutes > 0 ? `${Math.round((entrySummary.billableMinutes / entrySummary.totalMinutes) * 100)}% billable` : "0% billable"}`}
+                icon={Icon.Clock}
+                accessories={[
```
[the remark this answers is not in the document]
Comment (JuanVqz) [reply src/views/EntriesView.tsx]: Created a component for it
